### ISSUE TITLE
Trinkets Support [1.20.1]

### ISSUE THREAD

**Issue by otowm** [Thomilist/dimensional-inventories#6]
the itens in trinkets inventory dont disappear when passing the portals

**Comment by NoPyro** [comment]
This feature would also be nice for 1.19.2, trying to use Backpacked but it just remains in the inventory regardless, defeating the point.

**Comment by Thomilist** [comment]
I agree that modded inventory support would be valuable. Unfortunately, the mod currently suffers from being hacked together in a couple days and just polished slightly from there, meaning all the slots and effects that it saves and loads are hard-coded.

I'm tempted to scrap the current save/load implementation in favour of one that uses Minecraft's own, which it appears other mods tend to use as well. I'll have to do some digging and figure out how it works, but after that, it's just a matter of adding a trigger or two on dimension change. That should address not only Trinkets, but also Linked Storage (#5) and, I think, advancements (#4).

I assume automatic migration from the old to the new version would be a big plus as well. I can imagine manual migration being a bit of a pain if you have any more than a couple of players.

**Comment by LordMaximus05** [comment]
Has there been any update to this issue? I am also having issues with trinkets and backpack slots not clearing.

**Comment by Thomilist** [comment]
The status is that it is *expected* behaviour with the current releases, though obviously not *desirable*. It's the top priority of the project, but progress is unfortunately slow for a few reasons: "real life" obligations, my inexperience with Minecraft modding, and my desire to solve this properly rather than implementing a mod-specific patch (which would give me the same issue again whenever support for a new mod is wanted, or possibly even if a previously-supported mod changes in a certain way). Sorry to disappoint :/

**Comment by LordMaximus05** [comment]
no disappointment here! I know nothing about coding and so all of this is functionally magic to me lol take your time, its a very busy time of year and family/life comes first. But as a thought, (again I know nothing of coding) would it be possible to have like, a separate set of player data that is dimension specific? That way inventories, trinket slots, achievements etc are all separate.

**Comment by LordMaximus05** [comment]
Sorry for Necro posting but did this ever get looked into? I really want this to work since there is no other way to get a creative dimension.

**Comment by Thomilist** [comment]
No worries. The recent rewrite was partly to support features like this more easily. So, yeah, I haven't forgotten about it, but it's been a long time, so it's completely fair to ask.

Still for 1.20.1, right?

**Comment by LordMaximus05** [comment]
yes! Our server is 1.20.1 and we can't update to 1.20.4 because not all the mods are ready for that yet :)

**Comment by richardscull** [comment]
@Thomilist Any progress with this? Having creative lobby in different dimension becomes unreal when users can sneak items out using backpack - So your changes would mean a lot to me.

**Comment by Thomilist** [comment]
@richardscull I have started working on it a while ago, but long days at work and a few weeks of sickness (nothing concerning, but enough to keep me from doing much of anything) have slowed it down. Still, I *want* to do it, and I should have much better time over the holidays.

**Comment by Thomilist** [comment]
[Progress](https://github.com/Thomilist/dimensional-inventories-extension-trinkets) is being made (though more thorough testing is required before release).

**Comment by TheOneRingLOTR** [comment]
Hi there, I am quite interested in seeing this work as it's quite a necessity for my server right now. Will there be a 1.21.1 version as well? Thanks for your continued effort, your mod is truly much appreciated!

**Comment by Thomilist** [comment]
@TheOneRingLOTR Sure! Once the initial work is done for 1.20.1, creating a version for 1.21.1 should be relatively easy, so yes, there will be. And thank you for the kind words :)

**Comment by LordMaximus05** [comment]
Thank you so much for not abandoning this mod! I am so excited for this to work ^_^

**Comment by Thomilist** [comment]
Well this is a little bit exciting...

I think I have a working version! I've attached the mod files below - both the main mod and the extension mod that adds Trinkets support. I've tested it with [XP Storage Trinkets](https://modrinth.com/mod/xp-storage-trinkets/version/1.0+1.20.1) and [Elytra Slot](https://modrinth.com/mod/elytra-slot/version/7w60aZYA), but it *should* work with trinkets from any [Trinkets](https://modrinth.com/mod/trinkets/version/3.7.2)-compatible mod.

Let me know if it works with your setups as well. Oh, and probably **make a backup before trying it out**, because it also includes the major overhaul and storage format change from v1.x.x to v2.x.x. I don't *expect* the automatic migration to cause issues (especially since I've added an automated test specifically to cover migration), but a backup doesn't hurt :)

[dimensional-inventories-extension-trinkets-1.0.0-dev.1+1.20.1.zip](https://github.com/user-attachments/files/18724902/dimensional-inventories-extension-trinkets-1.0.0-dev.1%2B1.20.1.zip)

**Comment by LordMaximus05** [comment]
Hi, I tried to use the mod you linked but it still doesn't work :/ I am using Trinkets and Traveler's Backpacks and the backpacks are the main issue since people can load them up and travel back and forth. Do you have a discord server btw? I don't see one posted anywhere.

**Comment by Thomilist** [comment]
Thank you for giving it a try. I'll give it a closer look (probably in the weekend), but I have a guess as to what's going on...

Does the backpack item _itself_ get handled correctly? And if so, is the problem that players can simply equip a new backpack and access the same, shared backpack inventory across dimension pools? Because then it would seem that the backpack _inventory_ is not tied to the backpack _item_, but rather to the player. Not a showstopper, but it would require a bit of extra logic specifically for Traveler's Backpacks if that's the case.

And no, I do not have a Discord server, so that's why you can't find one :)

**Comment by LordMaximus05** [comment]
No, the backpack itself was not removed. I may have just done something wrong but the rest of my inventory was removed/restored like normal.

**Comment by Thomilist** [comment]
I'm afraid I have to give you the ol' *works on my machine*. Backpacks, including contents, are removed and restored as expected in my testing.

I tested on Minecraft 1.20.1 with Fabric Loader 0.16.10-1.20.1 and the following mods (also attached):

- `cloth-config-11.1.136-fabric`
- `dimensional-inventories-2.2.0-dev.1+1.20.1`
- `dimensional-inventories-extension-trinkets-1.0.0-dev.1+1.20.1`
- `fabric-api-0.92.3+1.20.1`
- `travelersbackpack-fabric-1.20.1-9.1.29`
- `trinkets-3.7.2`

[mods.zip](https://github.com/user-attachments/files/19436718/mods.zip)

Does it affect all types of backpacks?

**Comment by LordMaximus05** [comment]
Hmm, that's strange. I will do more testing tomorrow to do more testing. I'll keep you posted!

**Comment by LordMaximus05** [comment]
OMG IT WORKS!!!!! Thank you soooooooooo much!!!

**Comment by Thomilist** [comment]
Oh nice! Love to hear that.

I'll port it to a few versions, post them on Modrinth, and finally close this issue once that's done. I'm probably sticking to 1.20.1, 1.20.4 and 1.21.1 as "LTS" targets, as those seem to be the most popular 1.20+ versions to mod. Oh, and the latest release once Trinkets itself is available for that, of course.

Thank you very much for the input, feedback, testing, patience, and just the general tone :)

**Comment by loganhawker** [comment]
Hello! 

I noticed my trinkets disappearing too and decided to see if this had been brought up before. Excited to see it's already been addressed and you even shared a working version! Do you have an idea of when this would make it to Modrinth for 1.20.1?

Just dropped a sponsorship to thank you for making this incredibly useful mod!

**Comment by Thomilist** [comment]
Hello @loganhawker!

Thank you very much for both the kind words and the sponsorship(!), and sorry for the delayed response.

I've finished backporting the necessary changes to the base mod (Dimensional Inventories) for Minecraft `1.20.1`, `1.20.4` and `1.21.1`, and I've ported the Trinkets extension mod to those Minecraft versions as well. I've also uploaded releases to both GitHub and Modrinth, though [the Modrinth page for the Trinkets extension mod](https://modrinth.com/mod/dimensional-inventories-extension-trinkets) is still under review at this moment (which is to be expected).

With that done, I think I can finally close this issue :)

**Comment by loganhawker** [comment]
It works! Thank you!
